src/graphql/queries.js

```diff
@@ -169,68 +108,130 @@ fragment JobData on Job {
     }
   }
 }
-
-# WORKFLOW DATA END
 `
 
 /**
- * Query used to retrieve data for the application Dashboard.
- * @type {string}
+ * @type {DocumentNode}
  */
-const DASHBOARD_QUERY = `
-subscription DashboardSubscriptionQuery {
-  workflows {
-    id
-    name
-    status
+const WORKFLOW_TREE_DELTAS_SUBSCRIPTION = gql`
+subscription OnWorkflowTreeDeltasData ($workflowId: ID) {
+  deltas (workflows: [$workflowId], stripNull: true) {
+   ...WorkflowTreeDeltas
+  }
+}
+
+# TREE DELTAS BEGIN
+
+fragment WorkflowTreeDeltas on Deltas {
+  id
+  shutdown
+  added {
+    ...WorkflowTreeAddedData
+  }
+  updated {
+    ...WorkflowTreeUpdatedData
+  }
+  pruned {
+    ...WorkflowTreePrunedData
   }
 }
+
+fragment WorkflowTreeAddedData on Added {
+  workflow {
+    ...WorkflowData
+  }
+  cyclePoints: familyProxies (ids: ["root"], ghosts: true) {
+    ...CyclePointData
+  }
+  familyProxies (exids: ["root"], sort: { keys: ["name"] }, ghosts: true) {
+    ...FamilyProxyData
+  }
+  taskProxies (sort: { keys: ["cyclePoint"], reverse: false }, ghosts: true) {
+    ...TaskProxyData
+  }
+  jobs (sort: { keys: ["submit_num"], reverse:true }) {
+    ...JobData
+  }
+}
+
+fragment WorkflowTreeUpdatedData on Updated {
+  taskProxies (ghosts: true) {
+    ...TaskProxyData
+  }
+  jobs {
+    ...JobData
+  }
+  familyProxies (exids: ["root"], ghosts: true) {
+    ...FamilyProxyData
+  }
+}
+
+fragment WorkflowTreePrunedData on Pruned {
+  jobs
+  taskProxies
+  familyProxies
+}
+
+# TREE DELTAS END
+
+# WORKFLOW DATA BEGIN
+
+${WORKFLOW_DATA_FRAGMENT}
+
+${CYCLEPOINT_DATA_FRAGMENT}
+
+${FAMILY_PROXY_DATA_FRAGMENT}
+
+${TASK_PROXY_DATA_FRAGMENT}
+
+${JOB_DATA_FRAGMENT}
+
+# WORKFLOW DATA END
 `
 
 /**
  * Query used to retrieve data for the GScan sidebar.
- * @type {string}
+ * @type {DocumentNode}
  */
-const GSCAN_QUERY = `
+const GSCAN_DELTAS_SUBSCRIPTION = gql`
 subscription GscanSubscriptionQuery {
-  workflows {
-    id
-    name
-    status
-    owner
-    host
-    port
-    stateTotals
-    latestStateTasks(states: [
-      "failed",
-      "preparing",
-      "submit-failed",
-      "submitted",
-      "running"
-    ])
+  deltas (stripNull: true) {
+    ...GScanTreeDeltas
   }
 }
-`
 
-/**
- * Subscription used in the view that lists workflows in a table.
- * @type {string}
- */
-const WORKFLOWS_TABLE_QUERY = `
-subscription WorkflowsTableQuery {
-  workflows (ignoreInterval: 0) {
-    id
-    name
-    owner
-    host
-    port
+# GSCAN DELTAS BEGIN
+
+fragment GScanTreeDeltas on Deltas {
+  id
+  shutdown
+  added {
+    ...GscanAddedData
+  }
+  updated {
+    ...GscanUpdatedData
+  }
+  pruned {
+    workflow
   }
 }
+
+fragment GscanAddedData on Added {
+  workflow {
+    ...WorkflowData
+  }
+}
+
+fragment GscanUpdatedData on Updated {
+  workflow {
+    ...WorkflowData
+  }
+}
+
+${WORKFLOW_DATA_FRAGMENT}
 `
 
 export {
   WORKFLOW_TREE_DELTAS_SUBSCRIPTION,
-  DASHBOARD_QUERY,
-  GSCAN_QUERY,
-  WORKFLOWS_TABLE_QUERY
```

Comment:
Note to self: add the queries even though they won't result in a different `root` query?

Reply:
Dashboard added back.